swagger/src/main/scala/org/http4s/rho/swagger/TypeBuilder.scala

```diff
@@ -80,23 +82,64 @@ object TypeBuilder {
 
           models ++ generics ++ children
 
+        case tpe@TypeRef(_, sym: Symbol, tpeArgs: List[Type]) if isSumType(sym) =>
+          // TODO promote methods on sealed trait from children to model
+          modelToSwagger(tpe, sfs).map(addDiscriminator(sym)).toSet.flatMap { model =>
+            val parent = RefModel(model.id + "Ref", model.id2 + "Ref", model.id2)
+            val children =
+              sym.asClass.knownDirectSubclasses.flatMap( sub =>
+                  modelToSwagger(sub.asType.toType, sfs).map(composedModel(parent))
+                )
+            alreadyKnown ++ Set(model) ++ children
+          }
+
         case e =>
           Set.empty
       }
 
     go(t, alreadyKnown, known)
   }
 
+  private def addDiscriminator(sym: Symbol)(model: ModelImpl): ModelImpl = {
+    val typeVar = sym.annotations
+      .withFilter( a => a.tpe <:< typeOf[DiscriminatorField] )
+      .flatMap(a => a.tree.children.tail.collect { case Literal(Constant(field: String)) => field } )
+      .headOption.getOrElse("type")
```

Comment:
I think that there isn't a reasonable default for the discriminator field or else we wouldn't need the annotation. In that case it may be better to reject the model and log the reason for failure at info or maybe even warning level.